packages/openstef-models/src/openstef_models/models/forecasting/base_case_forecaster.py

```diff
@@ -0,0 +1,305 @@
+        ...         primary_lag=timedelta(days=7),
+        ...         fallback_lag=timedelta(days=21)
+        ...     )
+        ... )
+        >>> custom_forecaster = BaseCaseForecaster(custom_config)
+        >>>
+        >>> # Assumes data has lag columns like 'load_lag_-P7D', 'load_lag_-P14D'
+        >>> forecaster.fit(training_data)  # doctest: +SKIP
+        >>> predictions = forecaster.predict(test_data)  # doctest: +SKIP
+
+    Note:
+        The forecaster expects lag columns to be present in the input data, typically
+        created by applying LagTransform during preprocessing. The column names are
+        constructed as: {target_column}_lag_{ISO8601_duration}
+
+    See Also:
+        LagTransform: A transformation that creates lag features for time series forecasting.
+    """
+
+    _config: BaseCaseForecasterConfig
+
+    def __init__(
+        self,
+        config: BaseCaseForecasterConfig | None = None,
+    ) -> None:
+        """Initialize the base case forecaster.
+
+        Args:
+            config: Configuration specifying quantiles, horizons, and lag hyperparameters.
+                   If None, uses default configuration with 7-day primary and 14-day fallback lags.
+        """
+        self._config = config or BaseCaseForecasterConfig()
+        self._lag_columns_detected: bool = False
+        self._primary_lag_target_column_name: str | None = None
+        self._fallback_lag_target_column_name: str | None = None
+
+    @property
+    @override
+    def config(self) -> BaseCaseForecasterConfig:
+        return self._config
+
+    @property
+    @override
+    def hyperparams(self) -> BaseCaseForecasterHyperParams:
+        return self._config.hyperparams
+
+    @override
+    def to_state(self) -> State:
+        return {
+            "version": MODEL_CODE_VERSION,
+            "config": self.config.model_dump(mode="json"),
+        }
+
+    @override
+    def from_state(self, state: State) -> Self:
+        if not isinstance(state, dict) or "version" not in state or state["version"] > MODEL_CODE_VERSION:
+            raise ModelLoadingError("Invalid state for BaseCaseForecaster")
+
+        return self.__class__(config=BaseCaseForecasterConfig.model_validate(state["config"]))
+
+    @staticmethod
+    def _construct_lag_column_name(target_column: str, lag: timedelta) -> str:
+        """Constructs the column name for the lag target feature.
+
+        Args:
+            target_column: The name of the target column
+            lag: The timedelta lag
+
+        Returns:
+            The lag feature column name
+        """
+        # Convert timedelta to ISO format (e.g., P7D becomes -P7D for negative lag)
+        iso_lag = timedelta_to_isoformat(-lag)
+        return f"{target_column}_lag_{iso_lag}"
+
+    @property
+    @override
+    def is_fitted(self) -> bool:
+        return self._lag_columns_detected
+
+    def _detect_lag_columns(self, data: ForecastInputDataset) -> None:
```

Comment:
Perhaps we should lag the data ourselves in this model. It's hard to enforce that a certain transform should be used for a specific model. 

It's simpler to just lag the load from the context window since it's already properly incorporates availability constraints.